components/select/src/single-select-a11y/single-select-a11y.js

```diff
@@ -0,0 +1,392 @@
+import { requiredIf } from '@dhis2/prop-types'
+import { sharedPropTypes } from '@dhis2/ui-constants'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import i18n from '../locales/index.js'
+import { Menu } from './menu/index.js'
+import { SelectedValue } from './selected-value/index.js'
+import { optionProp } from './shared-prop-types.js'
+import {
+    useHandleKeyPress,
+    useHandleKeyPressOnFilterInput,
+} from './use-handle-key-press/index.js'
+
+function useFocussedOptionIndex({ filterable, filterValue, options }) {
+    const [defaultFocussedOptionIndex, setDefaultFocussedOptionIndex] =
+        useState(0)
+    const [searchFocussedOptionIndex, setSearchFocussedOptionIndex] =
+        useState(0)
+
+    // We want to reset the focussed option index when searching when the
+    // options change BUT we don't want to reset options when adding more
+    // options by using the "infinite scrolling" strategy
+    const initialized = useRef(false)
+    const prevFirstOption = useRef()
+    useEffect(() => {
+        // Ignore first call to prevent unnecessary re-render
+        if (!initialized.current) {
+            initialized.current = true
+            prevFirstOption.current = options[0]?.value
+        } else if (
+            filterable &&
+            prevFirstOption.current !== options[0]?.value
+        ) {
+            prevFirstOption.current = options[0]?.value
+            setSearchFocussedOptionIndex(0)
+        }
+    }, [options, filterable, initialized, prevFirstOption])
+
+    const focussedOptionIndex = useMemo(() => {
+        return filterValue
+            ? searchFocussedOptionIndex
+            : defaultFocussedOptionIndex
+    }, [defaultFocussedOptionIndex, searchFocussedOptionIndex, filterValue])
+
+    const setFocussedOptionIndex = useMemo(() => {
+        return filterValue
+            ? setSearchFocussedOptionIndex
+            : setDefaultFocussedOptionIndex
+    }, [filterValue])
+
+    return [focussedOptionIndex, setFocussedOptionIndex]
+}
+
+export function SingleSelectA11y({
+    options,
+    idPrefix,
+    onChange,
+    autoFocus = false,
+    className = '',
+    clearText: _clearText = '',
+    clearable = false,
+    customOption = undefined,
+    dataTest = 'dhis2-singleselecta11y',
+    dense = false,
+    disabled = false,
+    empty: _empty = false,
+    error = false,
+    filterHelpText = '',
+    filterLabel = '',
+    filterPlaceholder: _filterPlaceholder = '',
+    filterValue = '',
+    filterable = false,
+    labelledBy = '',
+    loading = false,
+    menuLoadingText: _menuLoadingText = '',
+    menuMaxHeight = '288px',
+    noMatchText: _noMatchText = '',
+    optionUpdateStrategy = 'polite',
+    placeholder = '',
+    prefix = '',
+    tabIndex = '0',
+    valid = false,
+    value = '',
```

Comment:
Would it make sense for the value to be an object with the same structure as the options? 
I feel like this would make it more consistent when passing the props.

Reply:
Yeah, that's actually a good idea.. Would get rid of the `valueLabel` prop, too!